File: Unidad 3-Funciones/Ejercicio11.lua
--[[
Autor: Guillermo Canto Dzul
Entradas: Monto a retirar del cajero
Salidas: Saldo luego del retiro
]]
--Funciones
function entrada()
    montoARetirar = io.read("*n")
end

function proceso()
  saldo = 3000
    if montoARetirar > 3000 or montoARetirar <= 0 then
        flag = 0
    else
        saldo = saldo - montoARetirar
        flag = 1
    end
end

function salida()
    if flag == 0 then
        io.write("La operacion no pudo realizarse")
    else
        io.write("Tu saldo es de " .. saldo )
    end
end
--Main
entrada()
proceso()
salida()
--[[
QA: Hector Abraham V. Cosgalla
Entradas:3000, 500, 4000
Salidas: Tu saldo es de .. saldo .. , Tu saldo es de .. saldo .. , La operacion no puede realizarse
Comentarios: Falta documentacion (corregido)
]]

File: Unidad 3-Funciones/Ejercicio11.c
#include <stdio.h>
/*Autor: Guillermo Canto Dzul
Entradas: Monto a retirar del cajero
Salidas: Saldo luego del retiro
*/
float entrada();
float proceso(float montoARetirar);
void salida(float saldo);
int main(int argc, char *argv[]){
	float montoARetirar, saldo;
	montoARetirar = entrada();
	saldo = proceso(montoARetirar);
	salida(saldo);
	return 0;
}
//Lee el monto a retirar
float entrada(){
	float montoARetirar;
	printf("Ingrese el monto a retirar:\n");
	scanf("%f", &montoARetirar);
	return montoARetirar;
}
//Calcula el saldo restante
float proceso(float montoARetirar){
	float saldo = 3000;
	if (montoARetirar > 3000 || montoARetirar <=0){
		printf("La operacion no pudo realizarse\n");
	}
	else{
		saldo = saldo - montoARetirar;
	}
	return saldo;
}
//imprime el saldo restante
void salida(float saldo){
	printf("Tu saldo es de %f\n", saldo);
}



